src/object/srv_ec_aggregate.c

```diff
@@ -856,9 +866,22 @@ agg_update_vos(struct ec_agg_entry *entry, bool write_parity)
 		recx.rx_idx = (entry->ae_cur_stripe.as_stripenum * len) |
 							PARITY_INDICATOR;
 		recx.rx_nr = len;
+		if (csummer != NULL) {
+			rc = daos_csummer_calc_iods(csummer, &sgl, &iod, NULL,
+						    1, false, NULL, 0,
+						    &iod_csums);
+			if (rc) {
+				D_ERROR("daos_csummer_calc_iods failed: "DF_RC
+					"\n", DP_RC(rc));
+				goto out;
+			}
+			D_ASSERT(iod_csums != NULL);
+		}
 		rc = vos_obj_update(ap->ap_cont_handle, entry->ae_oid,
 				    entry->ae_cur_stripe.as_hi_epoch, 0, 0,
-				    &entry->ae_dkey, 1, &iod, NULL, &sgl);
+				    &entry->ae_dkey, 1, &iod, iod_csums, &sgl);
+		if (csummer != NULL && iod_csums != NULL)
```

Comment:
don't need to check csummer anymore, (iod_csums != NULL) implies csummer != NULL